index.js

```diff
@@ -15,46 +39,135 @@ module.exports = function(parent, theirs, mine) {
+ * @param value - The current object value to diff
+ * @param {object} parent - Object with parent values
+ * @param {object} theirs - Object with their changes
+ * @param {object} mine - Object with my changes
+ * @param {array} path - Path to value in parent/theirs/mine
+ * @param {object} options - Key specific options for diff
+ * @returns {Array} - Array of objects specifying the differences
+ */
+function processKeyValuePair(key, value, parent, theirs, mine, path, options) {
+  var results = [];
+
+  // Only process keys that have no options, or have not been flagged as ignored
+  if (_.isUndefined(options[key]) || (!_.isUndefined(options[key]) && !options[key]['ignoreKey'])) {
+    if (!_.isArray(value) && !_.isObject(value)) {
       path.push(key);
-      var differences = compareValues(parent[key], theirs[key], mine[key], path);
-      if(differences) {
+      var differences = compareValues(parent[key], theirs[key], mine[key], path, options[key] || {});
+      if (differences) {
         results = results.concat(differences);
       }
       path.pop();
     }
-    else if(_.isArray(value)) {
-      // TODO
+    else if (_.isArray(value)) {
+      var array = value;
+      // If single dimension array run the comparison
+      if (value.filter(function (i) {
+          return !_.isArray(i) && !_.isObject(i);
+        }).length == value.length) {
+        path.push(key);
+        var differences = compareValues(parent[key], theirs[key], mine[key], path, options[key] || {});
+        if (differences) {
+          results = results.concat(differences);
+        }
+        path.pop(key);
+      }
+      _.forOwn(array, function (value, key) {
+        if (!_.isArray(value) && !_.isObject(value)) {
+
+        } else {
+          // TODO handle arrays of arrays or objects
+        }
+      });
     }
-    else if(_.isObject(value)) {
+    else if (_.isObject(value)) {
       path.push(key);
-      var differences = recurse(parent[key], theirs[key], mine[key], path);
-      if(differences) {
+      var differences = recurse(parent[key], theirs[key], mine[key], path, options[key] || {});
+      if (differences) {
         results = results.concat(differences);
       }
       path.pop();
     }
-  });
-
-  // TODO Handle all the keys that exist in parent but not mine
-  _.forOwn(parent, function(value, key) {
-
-  });
+  }
 
   return results;
 }
 
-function compareValues(parent, theirs, mine, path) {
-  if(_.isUndefined(parent) && _.isUndefined(theirs) && !_.isUndefined(mine)) {
+/**
+ * Compare the values in parent/theirs/mine to determine type of conflict/edit (if any)
+ *
+ * @param {object} parent - Object with parent values
+ * @param {object} theirs - Object with their changes
+ * @param {object} mine - Object with my changes
+ * @param {array} path - Path to value in parent/theirs/mine
+ * @param {object} options - Key specific options for diff
+ * @returns {*} - An object containing the conflict/edit, otherwise empty return
```

Comment:
Never actually seen `{*}` before, is that actually the standard? Versus something like `{} || null`